Fixed AArch32 compilation

blabug opened the pull request:
sockperf added support for AArch64, but at the same time it broke AArch32 support, as this wasn't catched by a compiler macro anymore and resulted in trying to assemble a rdtsc instruction.

See https://github.com/Mellanox/sockperf/commit/d84e8179f8ab007c7a19dfe263691b0429df7565 and https://github.com/Mellanox/sockperf/pull/187

Signed-off-by: Sven Püschel <s.pueschel@pengutronix.de>

swx-jenkins3 commented:
Can one of the admins verify this patch?

igor-ivanov commented on src/sockperf.cpp:
```diff
@@ -3903,6 +3911,7 @@ packet pace limit = %d",
         double ticks_per_second = (double)get_tsc_rate_per_second();
         log_dbg("+INFO: taking rdtsc directly consumes %.3lf nsec",
                 tdelta / SIZE * 1000 * 1000 * 1000 / ticks_per_second);
+#endif
```
Please consider using run-time logic in this place
```
if (!s_user_params.b_no_rdtsc) {
...
}
```

blabug replied:
fixed it in the last commit

igor-ivanov commented on src/sockperf.cpp:
```diff
@@ -2208,6 +2208,13 @@ static int parse_common_opt(const AOPT_OBJECT *common_obj) {
+        log_msg("AArch32 target build does not support rdtsc, use --no-rdtsc");
+        rc = SOCKPERF_ERR_BAD_ARGUMENT;
+    }
+#endif
```
Please instead of this check extend following configurable
https://github.com/Mellanox/sockperf/blob/sockperf_v2/src/sockperf.cpp#L2106

Something as
```
#if !defined(__windows__) && !(defined(__arm__) && !defined(__aarch64__))
#endif
```

blabug replied:
So instead of creating an error, the `b_no_rdtsc` value should be set by default for AArch32 targets?

Then I would have to alter the default value in https://github.com/Mellanox/sockperf/blob/sockperf_v2/src/defs.h#L740 , as skipping the argument parsing for AArch32 keeps the wrong value (using rdtsc)

igor-ivanov replied:
> So instead of creating an error, the `b_no_rdtsc` value should be set by default for AArch32 targets?
> 
yep, just set as default value `b_no_rdtsc=true` and remove usage of `--no-rdtsc` in case `(defined(__arm__) && !defined(__aarch64__))`
I missed this place https://github.com/Mellanox/sockperf/blob/sockperf_v2/src/sockperf.cpp#L269

> Then I would have to alter the default value in https://github.com/Mellanox/sockperf/blob/sockperf_v2/src/defs.h#L740 , as skipping the argument parsing for AArch32 keeps the wrong value (using rdtsc)

blabug replied:
removed the option when building for AArch32

igor-ivanov commented:
bot:retest

pslldq committed:
Fixed AArch32 compilation

sockperf added support for AArch64, but at the same time it broke AArch32 support,
as this wasn't catched by a compiler macro anymore and resulted in trying to assemble a rdtsc instruction.

See https://github.com/Mellanox/sockperf/commit/d84e8179f8ab007c7a19dfe263691b0429df7565
and https://github.com/Mellanox/sockperf/pull/187

Signed-off-by: Sven Püschel <s.pueschel@pengutronix.de>

igor-ivanov commented:
bot:retest
